### PR TITLE
Add get_sql_location for mobile workers

### PR DESCRIPTION
**Pull request by emord**
@esoergel http://manage.dimagi.com/default.asp?239572#1233519

buddy @czue

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -1623,6 +1623,10 @@ def sql_location(self):
                 pass
         return None
 
+    @memoized
+    def get_sql_location(self, domain):
+        return self.sql_location
```
Comment (czue) [inline corehq/apps/users/models.py]: @esoergel does this seem like the right approach?

**Review by czue — APPROVED** [review]

**Comment by esoergel** [comment]
This seems fine as a quickfix approach.  It's more concerning that we have mobile workers making requests that we'd expect to come from web users.  Should `CommCareUser` be a subclass of `WebUser`?  I imagine we must have a bunch more of these incompatibilities.

**Comment by emord** [comment]
fwiw I was only able to reproduce this issue when I assigned a mobile worker to a role that could edit mobile workers. There's a select2 on the mobile worker edit page.

It seems like a weird thing to do, but it's not something we specifically prohibit

**Comment by czue** [comment]
> Should `CommCareUser` be a subclass of `WebUser`? 

I think this would probably create more confusion as they are fundamentally different in different ways (most notably in terms of domain membership) so would rather address whatever calling code doesn't realize that these are `CouchUser`s instead of the subclasses.

**Comment by czue** [comment]
can we move the function up to `CouchUser`?

**Comment by emord** [comment]
:white_check_mark:

**Comment by esoergel** [comment]
@emord FYI Cory had marked this PR as approved, so it's legit for you to self-merge after the build passes.

**Pull request merged**